llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp

```diff
@@ -1924,8 +1924,9 @@ void X86MCCodeEmitter::encodeInstruction(const MCInst &MI,
   if ((TSFlags & X86II::OpMapMask) == X86II::ThreeDNow)
     emitByte(X86II::getBaseOpcodeFor(TSFlags), CB);
 
-  assert(CB.size() - StartByte <= 15 &&
-         "The size of instruction must be no longer than 15.");
+  if (CB.size() - StartByte > 15)
+    Ctx.reportError(MI.getLoc(),
```

Comment:
https://llvm.org/docs/CodingStandards.html#error-and-warning-messages

"the size ... " (remove trailing period)

Reply:
Done

Reply:
gas has a slightly different diagnostic. Consider matching it? 
https://sourceware.org/bugzilla/show_bug.cgi?id=31323